src/plopp/backends/pythreejs/mesh3d.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+import uuid
+from typing import Literal
+
+import numpy as np
+import scipp as sc
+from matplotlib.colors import to_rgb
+
+from ...core.limits import find_limits
+from ...graphics.bbox import BoundingBox
+from .canvas import Canvas
+
+
+class Mesh3d:
+    """
+    Artist to represent a three-dimensional mesh.
+
+    Parameters
+    ----------
+    canvas:
+        The canvas to draw the mesh on.
+    data:
+        The initial data to create the mesh from. Must be a DataGroup that contains at
+        least the following fields:
+        - vertices: a DataArray with the vertices of the mesh.
+        - faces: a DataArray with the faces of the mesh.
+    color:
+        The color of the mesh. If None, the mesh will be colored according to the
+        artist number.
+    opacity:
+        The opacity of the mesh.
+    edgecolor:
+        The color of the edges of the mesh. If None, the edges will not be shown.
+    artist_number:
+        The number of the artist. This is used to determine the color of the mesh if
+        `color` is None.
+    """
+
+    def __init__(
+        self,
+        *,
+        canvas: Canvas,
+        data: sc.DataArray,
+        color: str | None = None,
+        opacity: float = 1,
+        edgecolor: str | None = None,
+        artist_number: int = 0,
+    ):
+        import pythreejs as p3
+
+        self._data = data
+        self._canvas = canvas
+        self._artist_number = artist_number
+        self._id = uuid.uuid4().hex
+
+        # Note: index *must* be unsigned!
+        index = p3.BufferAttribute(
+            array=self._data.coords["faces"]
+            .value.values.flatten()
+            .astype('uint32', copy=False)
+        )
+
+        attributes = {
+            'position': p3.BufferAttribute(
+                array=np.array(
+                    [
+                        self._data.coords["x"].values.astype('float32', copy=False),
+                        self._data.coords["y"].values.astype('float32', copy=False),
+                        self._data.coords["z"].values.astype('float32', copy=False),
+                    ]
+                ).T
```

Comment:
Don't know what size of mesh this can or should support. But you could avoid extra copies here by not taking x,y,z apart into separate variables. Instead, you could use
```python
p3.BufferAttribute(positions.values.astype('float32', copy=False))
```
(`copy=False` doesn't do anything here because the positions vector is always `float64`.)

Reply:
I split up the coordinates because I wanted the [figure](https://github.com/scipp/plopp/pull/376/files#diff-b9679c47242db4cccfbd303f917abb03780e549cfaf786c5f0bdbcb8682da0f3R81) to behave a bit like `scatter3d`, with checking the units of the x,y,z coordinates. But maybe I can make something that just takes in a `vertices` coordinate...